test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationParts/AssemblyPartTest.cs

```diff
@@ -22,6 +22,20 @@ public void AssemblyPart_Name_ReturnsAssemblyName()
         }
 
         [Fact]
+        public void AssemblyPart_Types_ReturnsDefinedTypes()
+        {
+            // Arrange
+            var assembly = typeof(AssemblyPartTest).GetTypeInfo().Assembly;
+            var part = new AssemblyPart(assembly);
+
+            // Act
+            var types = part.Types;
+
+            // Assert
+            Assert.Equal(assembly.DefinedTypes, types);
```

Comment:
BTW I'd argue that this test is testing the specific implementation rather than the behavior. It would be safer to assert that the collections are equal, rather than checking that the collection *instance* is the same.